[quant] Adding dynamic qRNN

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36097 [quant] Adding dynamic qRNN**

Differential Revision: [D20878115](https://our.internmc.facebook.com/intern/diff/D20878115)

Commit:
[quant] Adding dynamic qRNN

Commit:
Update on "[quant] Adding dynamic qRNN"


Differential Revision: [D20878115](https://our.internmc.facebook.com/intern/diff/D20878115)

Commit:
Update on "[quant] Adding dynamic qRNN"


Differential Revision: [D20878115](https://our.internmc.facebook.com/intern/diff/D20878115)

Commit:
Update on "[quant] Adding dynamic qRNN"


Differential Revision: [D20878115](https://our.internmc.facebook.com/intern/diff/D20878115)

raghuramank100 commented on torch/nn/quantized/dynamic/modules/rnn.py:
```diff
@@ -308,30 +311,24 @@ def process_weights(ihhh, layer, suffix, dtype):
-                            weight.float(), float(wt_scale), int(wt_zp), torch.qint8)
-                        packed_weight = \
-                            torch.ops.quantized.linear_prepack(qweight, bias)
+                        qweight = _quantize_weight(weight, weight_observer)
```
Need to observe weights prior to calling quantize_weight
``` python
weight_observer(weight)
qweight = _quantize_weight(weight, weight_observer)
```

raghuramank100 replied:
We also need tests for these changes

z-a-f replied:
I agree -- I will send tests as well

Commit:
Update on "[quant] Adding dynamic qRNN"


Differential Revision: [D20878115](https://our.internmc.facebook.com/intern/diff/D20878115)

Commit:
Update on "[quant] Adding dynamic qRNN"


Differential Revision: [D20878115](https://our.internmc.facebook.com/intern/diff/D20878115)

Commit:
Update on "[quant] Adding dynamic qRNN"


Differential Revision: [D20878115](https://our.internmc.facebook.com/intern/diff/D20878115)

[ghstack-poisoned]
